### PR TITLE
fix: Reduce code complexity

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

File: src/backend/llvm/codegen/common.h [hunk 1]
```diff
@@ -37,6 +33,24 @@ ::llvm::Constant *getFPConstant<float>(::llvm::LLVMContext &ctx, float value);
...
 ::llvm::Constant *getFPConstant<double>(::llvm::LLVMContext &ctx, double value);
 
+using BuiltinThreeTensorArgs = std::function<void(::llvm::LLVMContext &,
+                                                  ::llvm::Module &,
+                                                  ::llvm::IRBuilder<> &,
+                                                  core::inner::Tensor &,
+                                                  core::inner::Tensor &,
+                                                  core::inner::Tensor &)>;
+
+using BuiltinTATATArgs = std::function<void(::llvm::LLVMContext &,
```
Comment (masdevas) [inline src/backend/llvm/codegen/common.h]: Sho?

File: src/backend/llvm/codegen/register_default_functors.cpp [hunk 2]
```diff
@@ -17,11 +17,11 @@
...
 namespace athena::backend::llvm::codegen {
 void registerDefaultFunctors(LLVMGenerator *generator) {
-    registerAdd(generator);
+    registerStandardBuiltin<BuiltinThreeTensorArgs>("add", generator);
     registerAllocate(generator);
     registerFill(generator);
-    registerHadamard(generator);
-    registerFma(generator);
-    registerMse(generator);
+    registerStandardBuiltin<BuiltinTATATArgs>("hadamard", generator);
```
Comment (masdevas) [inline src/backend/llvm/codegen/register_default_functors.cpp]: Sho?

File: src/backend/llvm/codegen/register_default_functors.cpp [hunk 3]
```diff
@@ -17,11 +17,11 @@
...
+    registerStandardBuiltin<BuiltinTATATArgs>("fma", generator);
```
Comment (masdevas) [inline src/backend/llvm/codegen/register_default_functors.cpp]: Sho?

File: src/backend/llvm/codegen/util.cpp [hunk 4]
```diff
@@ -24,4 +25,150 @@ ::llvm::Constant *getFPConstant<double>(::llvm::LLVMContext &ctx,
...
+/**
+ * Register builtin with three Tensor args biolerplate routine
+ * @param name Name of builtin
+ * @param generator LLVMGenerator instance
+ */
+template <>
+void registerStandardBuiltin<BuiltinThreeTensorArgs>(const std::string &name,
+                                                     LLVMGenerator *generator) {
+    BuiltinThreeTensorArgs f =
+        [generator, name](::llvm::LLVMContext &ctx, ::llvm::Module &module,
```
Comment (masdevas) [inline src/backend/llvm/codegen/util.cpp]: Are you actually want to copy "generator" and "name" by value?

Comment (alexbatashev) [reply src/backend/llvm/codegen/util.cpp]: Generator is a pointer and name is a const reference. Passing by reference has no effect here, I think.

File: src/backend/llvm/codegen/register_default_functors.cpp (hunk 2 above)
Comment (alexbatashev) [reply src/backend/llvm/codegen/register_default_functors.cpp]: src/backend/llvm/codegen/util.cpp:140

File: src/backend/llvm/codegen/common.h (hunk 1 above)
Comment (alexbatashev) [reply src/backend/llvm/codegen/common.h]: src/backend/llvm/codegen/util.cpp:140

File: src/backend/llvm/codegen/register_default_functors.cpp (hunk 3 above)
Comment (alexbatashev) [reply src/backend/llvm/codegen/register_default_functors.cpp]: src/backend/llvm/codegen/util.cpp:140

File: src/backend/llvm/codegen/util.cpp (hunk 4 above)
Comment (masdevas) [reply src/backend/llvm/codegen/util.cpp]: "name" will be copied

**Pull request merged**